api/v1/controllers/AuthController.js

```diff
@@ -0,0 +1,68 @@
+var Promise = require('bluebird');
+
+var errors = require('../errors');
+var AuthService = require('../services/AuthService');
+var UserService = require('../services/UserService');
+
+var router = require('express').Router();
+
+function _issueByEmail(email, password) {
+	if (!password && password !== null) {
+		throw new TypeError("To issue a token by email without checking the " +
+			"user's password, you must pass the password as the null type");
+	}
+
+	return UserService
+		.findUserByEmail(email)
+		.then(function (user) {
+			if (!password)
+				return AuthService.issueForUser(user);
+
+			return UserService
+				.verifyPassword(user, password)
+				.then(function () {
+					return AuthService.issueForUser(user);
+				});
```

Comment:
In some situations (such as a user's initial request for a token), we have no way of knowing that the user is making a valid claim except for his/her username and password. As such, we need to verify the username and password as well. 

In other situations (such as the user's request for a refresh token), we can verify the user's claim and issue a new token without requiring a password. That's when the password field is allowed to be null.

Maybe I should just refactor that method. I was trying to save myself from typing a bit, and I think that made things unclean.